contracts/CodexTitleMetadata.sol

```diff
@@ -48,22 +49,61 @@ contract CodexTitleMetadata is ERC721Token {
     return imageHashes[_index];
   }
 
-  function modifyDescriptionHash(uint256 _tokenId, bytes32 _newDescriptionHash) public canModify(_tokenId) {
-    tokenData[_tokenId].descriptionHash = _newDescriptionHash;
+  /**
+   * @dev Updates token metadata hashes to whatever is passed in
+   * @param _providerId (optional) An ID that identifies which provider is
+   *  minting this token
+   * @param _providerMetadataId (optional) An arbitrary provider-defined ID that
+   *  identifies the metadata record stored by the provider
+   */
+  function modifyMetadataHashes(
+    uint256 _tokenId,
+    bytes32 _newNameHash,
+    bytes32 _newDescriptionHash,
+    bytes32[] _newImageHashes,
+    string _providerId, // TODO: convert to bytes32?
+    string _providerMetadataId // TODO: convert to bytes32?
+  )
+    public canModify(_tokenId)
+  {
 
-    emit Modified(msg.sender, _tokenId, ModifiedType.DESCRIPTION_CHANGE);
-  }
+    require(exists(_tokenId));
 
-  function modifyNameHash(uint256 _tokenId, bytes32 _newNameHash) public canModify(_tokenId) {
-    tokenData[_tokenId].nameHash = _newNameHash;
+    // nameHash is only overridden if it's not a blank string, since name is a
+    //  required value
+    //
+    // @dev: is this the best way to check for an empty bytes32 array?
+    //  would (_newNameHash != "") be better in any way?
+    //  see: https://ethereum.stackexchange.com/questions/27227/why-does-require-length-of-bytes32-0-not-work
+    if (_newNameHash[0] != 0) {
+      tokenData[_tokenId].nameHash = _newNameHash;
+    }
 
-    emit Modified(msg.sender, _tokenId, ModifiedType.NAME_CHANGE);
-  }
+    // descriptionHash can always be overridden since it's an optional value
+    //  (e.g. you can "remove" a description by setting it to a blank string)
+    tokenData[_tokenId].descriptionHash = _newDescriptionHash;
 
-  function addNewImageHash(uint256 _tokenId, bytes32 _imageHash) public canModify(_tokenId) {
-    tokenData[_tokenId].imageHashes.push(_imageHash);
+    // imageHashes is only overridden if it has more than one value, since at
+    //  least one image (i.e. mainImage) is required
+    //
+    // @dev: is this the best way to check for an empty bytes32 array?
```

Comment:
Leave a TODO for these instead of @dev

Reply:
Weird, I actually did change these to TODOs at one point but they must have gotten reverted during some undos.